optuna/importance/_base.py

```diff
@@ -1,18 +1,23 @@
 from __future__ import annotations
 
 import abc
-from collections.abc import Callable
-from collections.abc import Collection
 from typing import cast
+from typing import TYPE_CHECKING
 
 import numpy as np
 
-from optuna._transform import _SearchSpaceTransform
-from optuna.distributions import BaseDistribution
-from optuna.search_space import intersection_search_space
-from optuna.study import Study
-from optuna.trial import FrozenTrial
-from optuna.trial import TrialState
+from optuna.search_space import intersection_search_space  # Used at runtime too
+from optuna.trial import TrialState  # Used at runtime, so keep it outside TYPE_CHECKING
```

Comment:
These comments are not necessary as this is what the `TYPE_CHECKING` statement is for.

```suggestion
from optuna.search_space import intersection_search_space
from optuna.trial import TrialState
```

Reply:
Updated as discussed.